default/AI/ShipDesignAI.py

```diff
@@ -1006,7 +1016,7 @@ def add_design(self, verbose=False):
         else:
             print "Failure: Tried to add design %s but returned %s, expected 1" % (design_name, res)
             return None
-        new_id = _get_design_by_name(design_name).id
+        new_id = _get_design_by_name(design_name,update_invalid=True).id
```

Comment:
This line seems to have been wrong...

If we did not get a design by this call, then this would throw an error as ```None``` obviously has no ```.id```... Obviously never happened and shouldn't happen as we just added that design, but still...
Also next line, id could in principle be 0 if no premade designs exist.

Would you mind fixing that in this PR?

```python
new_design = _get_design_by_name(design_name,update_invalid=True)
if new_design:
     Cache.map_reference_design_name[reference_name] = design_name
     return new_design.id
```